test/factories/cluster_group.rb

```diff
@@ -24,11 +24,20 @@
         annotation_input {
           []
         }
+        # range input is min/max values for each domain (e.g. total extent of plot area)
+        # populate raw values in from study_file, and then call compact/delete_if to remove nil values
+        range_input {
+          {
+              x: [study_file.x_axis_min, study_file.x_axis_max].compact,
+              y: [study_file.y_axis_min, study_file.y_axis_max].compact,
+              z: [study_file.z_axis_min, study_file.z_axis_max].compact
+          }.delete_if {|k, v| v.empty?}
+        }
```

Comment:
Since compact will remove the `nil` objects, by checking the array is then `empty?` later will remove them entirely.  This way we will only store values if provided, and if not, the entire `domain_range` object is `nil`.  If we don't do this, even with `nil` values the cluster "thinks" it has a range just based on the presence of the empty hash.

Having said all that, I'll try to figure out a cleaner way of doing this.

Reply:
b573c6a